scylla/src/transport/session.rs

```diff
@@ -865,27 +865,19 @@ impl Session {
 
         // Prepare statements on all connections concurrently
         let handles = connections.iter().map(|c| c.prepare(&query));
-        let mut results = join_all(handles).await;
+        let mut results = join_all(handles).await.into_iter();
 
-        // If at least one prepare was successful prepare returns Ok
+        // If at least one prepare was successful, `prepare()` returns Ok.
+        // Find first result that is Ok, or Err if all failed.
 
-        // Find first result that is Ok, or Err if all failed
-        let mut first_ok: Option<Result<PreparedStatement, QueryError>> = None;
-
-        while let Some(res) = results.pop() {
-            let is_ok: bool = res.is_ok();
-
-            first_ok = Some(res);
-
-            if is_ok {
-                break;
-            }
-        }
-
-        let mut prepared: PreparedStatement = first_ok.unwrap()?;
+        // Safety: there is at least one node in the cluster, so there will be at least
```

Comment:
I would clarify that `self.cluster.get_working_connections()?` returns at least one connection or an error. None empty cluster is not a good guarantee.